admin/javascript/src/sections/campaign-admin/controller.js

```diff
@@ -19,8 +19,23 @@ class CampaignAdminContainer extends SilverStripeComponent {
   }
 
   componentDidMount() {
-    window.ss.router(`/${this.props.config.campaignViewRoute}`, (ctx) => {
+    window.ss.router(`/${this.props.config.campaignViewRoute}`, (ctx, next) => {
       this.props.actions.showCampaignView(ctx.params.id, ctx.params.view);
+
+      // If we are coming from a non-campaigns section, we need to call next()
+      // to trigger an ajax reload
+
+      // @todo - Use state to detect section changed behaviour rather than referer
```

Comment:
I wonder if we can use https://github.com/visionmedia/page.js/#pageexitpath-callback-callback- instead, when we enter a campaign section we could set a variable to true, and set it to false again on exit.